storage-operator/pkg/apis/storage/v1alpha1/volume_types.go

```diff
@@ -66,6 +148,25 @@ type Volume struct {
 	Status VolumeStatus `json:"status,omitempty"`
 }
 
+// Return the volume size.
+func (self *Volume) GetSize(disk_size resource.Quantity) resource.Quantity {
+	if self.Spec.SparseLoopDevice != nil {
+		return self.Spec.SparseLoopDevice.Size
+	}
+	// RawBlockDevice use the whole disk, so return the disk size as size.
+	return disk_size
+}
+
+// Return the volume path on the node.
+func (self *Volume) GetPath() string {
+	if self.Spec.SparseLoopDevice != nil {
+		return fmt.Sprintf(
+			"/var/lib/metalk8s/storage/sparse/%s", self.UID,
+		)
+	}
+	return self.Spec.RawBlockDevice.DevicePath
```

Comment:
You check for `SparseLoopDevice` to be set above, but that not to be set does not imply `RawBlockDevice` *is* set (though it should be, but is this enforced somewhere?). As such, this API should return either a valid result, by checking every volume type, or fail with an error if no volume type is actually configured.

Reply:
I have this piece of code in `handleNewVolume`

```go
	// Check if a type is specified.
	if volume.Spec.SparseLoopDevice == nil &&
		volume.Spec.RawBlockDevice == nil {
		return self.setVolumeFailed(
			volume,
			storagev1alpha1.InternalError,
			"volume type not found in spec",
		)
	}
```
But I'll double check if it's enough to have it here, or if I need to move it elsewhere to be safe.

BTW: do we want to be safe against people tampering the volume (edit manually the CR to remove those or the status)?